crashreports/internal/win32/CrashReportContext.cpp

```diff
@@ -0,0 +1,115 @@
+#include <locale>
+#include <codecvt>
+#include <sstream>
+#include "client/windows/handler/exception_handler.h"
+#include "CrashReportContext.h"
+
+namespace {
+	bool uploadReport(
+		const wchar_t* dump_path,
+		const wchar_t* minidump_id,
+		void* context,
+		EXCEPTION_POINTERS* exinfo,
+		MDRawAssertionInfo* assertion,
+		bool succeeded)
+	{
+		CrashReportContext* crashReportContext = reinterpret_cast<CrashReportContext*>(context);
+		if (!crashReportContext->send(dump_path, minidump_id))
+			return false;
+		return succeeded;
+	}
+
+	std::string stringifyParameters(const std::map<std::string, std::string>& parameters)
```

Comment:
This and the remaining functions do not operate under the constraints of uploadReport().  Say that too in comments.